server/routers/api.ts

```diff
@@ -24,53 +24,188 @@ interface ManifestAccessTicket {
 }
 const manifestAccessTickets: Map<string, ManifestAccessTicket> = new Map();
 
+function applyLogFallbacks(logMessage: types.LogMessageSimple, defaults: Partial<types.LogMessage>) {
+  if (logMessage.entity) {
+    logMessage.entity.detail = {...(defaults.entity?.detail ?? {}), ...(logMessage.entity?.detail ?? {})}; 
+  }
+  logMessage.entity = {...defaults.entity, ...logMessage.entity};
+  logMessage.source = {...defaults.source, ...logMessage.source};
+  logMessage.agent = {...defaults.agent, ...logMessage.agent};
+  return {...defaults, ...logMessage};
+}
+
+function log(context: oak.Context, msg: types.LogMessageSimple, shl?: types.HealthLink) {  
+  let logMessage: types.LogMessage = {
+    version: "2",
+    severity: "info",
+    action: msg.action,
+    occurred: new Date().toISOString(),
+    subject: shl?.userId,
+    agent: {
+      ip_address: context.request.ip,
```

Comment:
@daniellrgn I'm finding that this is showing an internal network IP address (172.18.0.6). I had this same issue in cpro, @ivan-c  suggested apache's HTTP_X_REAL_IP, which worked there. I don't know if you need something specific to traefik, or... I see some references to `X-Forwarded-For`  https://stackoverflow.com/a/66003548 , and `X-Real-IP`... I don't see this as a blocker for today's release.

Reply:
Fixed this in the most recent commit here, just had to flip a proxy setting. Now seeing my IP:
`"agent":{"ip_address":"10.102.36.230",`